### PR TITLE
Tgbot 43 create object provider for engineer objects

### PR DESCRIPTION
**Pull request by AlexanderYarokhovich**
Create class that will provide 3 engineer objects.
https://schneidervs.atlassian.net/browse/TGBOT-43

### CODE REVIEW

**Review by schneidervs — DISMISSED** [review]

**Review by yurasava — DISMISSED** [review]

File: stoservice/src/main/java/com/tutrit/stoservice/provider/EngineerProvider.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.tutrit.stoservice.bean.Engineer;
+import com.tutrit.stoservice.repository.EngineerRepository;
+
+public class EngineerProvider {
+    Engineer engineer;
+    EngineerRepository engineerRepository;
+
+    public Engineer[] getEngineers() {
+        engineerRepository = new EngineerRepository();
```
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/provider/EngineerProvider.java:11]: provider should only create objects, not store them.

File: stoservice/src/test/java/com/tutrit/stoservice/provider/EngineerProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Test
+    void getEngineers() {
+        Engineer[] engineersActual = engineerProvider.getEngineers();
+        Engineer[] engineersExpected = new Engineer[engineersActual.length];
+        engineersExpected[0] = new Engineer("1","1","1","1","1","1",1,1);
+        engineersExpected[1] = new Engineer("2","2","2","2","2","2",2,2);
+        engineersExpected[2] = new Engineer("3","3","3","3","3","3",3,3);
+
+        for (int i =0; i < engineersActual.length; i++) {
```
Comment (max-java) [inline stoservice/src/test/java/com/tutrit/stoservice/provider/EngineerProviderTest.java:25]: usually we don't use cycles and decision operators in tests.

File: stoservice/src/main/java/com/tutrit/stoservice/bean/Engineer.java [hunk 3]
```diff
@@ -95,7 +95,7 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         Engineer engineer = (Engineer) o;
-        return experience == engineer.experience && generalExperience == engineer.generalExperience && Objects.equals(idEngineer, engineer.idEngineer) && Objects.equals(firstName, engineer.firstName) && Objects.equals(lastName, engineer.lastName) && Objects.equals(function, engineer.function) && Objects.equals(category, engineer.category) && Objects.equals(education, engineer.education);
+        return experience == engineer.experience && generalExperience == engineer.generalExperience && idEngineer.equals(engineer.idEngineer) && firstName.equals(engineer.firstName) && lastName.equals(engineer.lastName) && function.equals(engineer.function) && category.equals(engineer.category) && education.equals(engineer.education);
```
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/bean/Engineer.java:98]: leave that generated

File: stoservice/src/main/java/com/tutrit/stoservice/provider/EngineerProvider.java (hunk 1 above)
Comment (max-java) [reply stoservice/src/main/java/com/tutrit/stoservice/provider/EngineerProvider.java:11]: add int i to method paramethers

File: stoservice/src/test/java/com/tutrit/stoservice/provider/EngineerProviderTest.java (hunk 2 above)
Comment (max-java) [reply stoservice/src/test/java/com/tutrit/stoservice/provider/EngineerProviderTest.java:25]: use Assertions.assertArrayEquals

**Review by max-java — DISMISSED** [review]

**Review by yurasava — APPROVED** [review]
lgfm

**Review by Sav4ykVladzislau — APPROVED** [review]

**Comment by schneidervs** [comment]
- [ ] LGFM

**Review by schneidervs — APPROVED** [review]

**Review by lamaconda — APPROVED** [review]
lgfm

**Pull request merged**